### ISSUE TITLE
Single-sided mods

### ISSUE THREAD

**Issue by LoganDark** [Prospector/ModMenu#361]
Hello, I'm the developer and maintainer of a mod called [Fabric Console](https://github.com/LoganDark/fabric-console). I'm currently bringing the project up to date, which means I've updated Mod Menu to the latest 1.16 version (for now), 1.16.22.

It seems that the mod now lacks the option to specify a mod as client-side-only if it also supports the server, which _does_ make sense. However, Fabric Console _is_ purely a single-sided mod - it adds no new content, and simply allows for color in the console (dedicated server or client alike). In this case I still feel it's important to be able to articulate the fact that it is not required for both server & client to have this mod installed.

I propose a new custom field (name TBD) to specify that a mod requires no interaction between client & server. This would replace the "modmenu:clientsideOnly" field that is being deprecated & removed soon.

**Comment by K0-RR** [comment]
I'm not a dev and I might be wrong but according to the wiki ModMenu now uses https://github.com/TerraformersMC/ModMenu/blob/1.17/src/main/resources/fabric.mod.json#L6

> There are more badges, such as the Client badge, but this is automatically determined by the environment defined in the official fabric.mod.json metadata. Only the ones above can be manually assigned in the badges array.

**Comment by LoganDark** [comment]
> I'm not a dev and I might be wrong but according to the wiki ModMenu now uses https://github.com/TerraformersMC/ModMenu/blob/1.17/src/main/resources/fabric.mod.json#L6
> 
> > There are more badges, such as the Client badge, but this is automatically determined by the environment defined in the official fabric.mod.json metadata. Only the ones above can be manually assigned in the badges array.

Yes, I am pointing this out and saying that it is not adequate in this case. Fabric Console can run on either client or server, and it is completely independant that way. I'm asking for a badge or something that I can use to specify this. I (ab)used `modmenu:clientsideOnly` in the past, but that's deprecated.

When used on a client, it is a client side-only mod. When used on a server, it is server side-only.

**Comment by jackassmc** [comment]
This is a valid use case and should be supported. Thank you for raising this issue!

It might be possible to achieve this by specifying the environment as `client and server` instead of `*` and checking that the environment is client (current behavior) or contains client. I'll look into it.

**Comment by jackassmc** [comment]
The [fabric.mod.json v1 spec](https://fabricmc.net/wiki/documentation:fabric_mod_json_spec#optional_fields_mod_loading) allows a string or an array of strings but Fabric Loader 0.14.8 does not support an array of strings, it only supports `*`, `client`, and `server` (see [here](https://github.com/FabricMC/fabric-loader/blob/0.14.8/src/main/java/net/fabricmc/loader/impl/metadata/V1ModMetadataParser.java#L129) and [here](https://github.com/FabricMC/fabric-loader/blob/d713b2a801ad142394ac0eb1d1eaa452fc783021/src/main/java/net/fabricmc/loader/impl/metadata/V1ModMetadataParser.java#L249)).
